staging/src/k8s.io/sample-controller/artifacts/examples/main.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//This is an example of polymorphic scale client.
+
+package main
+
+import (
+	"flag"
+	"os"
+
+	"github.com/golang/glog"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/scale"
+	"k8s.io/client-go/tools/clientcmd"
+	clientset "k8s.io/sample-controller/pkg/client/clientset/versioned"
+)
+
+const (
+	namespace       = "default"
+	resourceName    = "example-foo"
+	resourceGroup   = "samplecontroller.k8s.io"
+	resourceVersion = "v1alpha1"
+)
+
+func main() {
+	err := start()
+	if err != nil {
+		glog.Fatalf("error: %v", err)
+		os.Exit(1)
+	}
+
+}
+func start() error {
+	var cfg *string
+	if c := kubeconfig(); c != "" {
+		cfg = flag.String("kubeconfig", c, "(optional) KUBECONFIG env of the kubeconfig file")
+	} else {
+		cfg = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
```

Comment:
Can we use the config loading/flags pattern from the main sample-controller here? https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/sample-controller/main.go#L46